Src/AutoFixtureUnitTest/Kernel/TerminatingWithPathSpecimenBuilderTest.cs

```diff
@@ -162,35 +159,62 @@ public void SpecimenRequestsWillInitiallyBeEmpty()
         [Fact]
         public void SpecimenRequestsRaisedFromTracerAreRecordedCorrectly()
         {
-            // Fixture setup
-            var tracer = new DelegatingTracingBuilder();
-            var specimens = new[] { new object(), new object(), new object() };
-            var requestEvents = specimens.Select((o, i) => new RequestTraceEventArgs(o, i)).ToList();
-            var sut = new TerminatingWithPathSpecimenBuilder(tracer);
-            // Exercise system
-            requestEvents.ForEach(tracer.RaiseSpecimenRequested);
-            // Verify outcome
-            Assert.True(specimens.SequenceEqual(sut.SpecimenRequests));
-            // Teardown
+            // arrange
+            var requests = new[] { new object(), new object(), new object() };
+            var requestQueue = new Queue<object>(requests);
+            var firstRequest = requestQueue.Dequeue();
+
+            object[] capturedRequests = null;
+            TerminatingWithPathSpecimenBuilder sut = null;
+            
+            var builder = new DelegatingSpecimenBuilder
+            {
+                OnCreate = (r, c) =>
+                {
+                    if(requestQueue.Count > 0) 
+                        return c.Resolve(requestQueue.Dequeue());
+                    
+                    capturedRequests = sut.SpecimenRequests.ToArray();
+                    return new object();
+                }
+            };
+            
+            sut = new TerminatingWithPathSpecimenBuilder(builder);
+            // Cause sut to be executed recursively for multiple times.
+            var context = new SpecimenContext(sut);
+            
+            //act
+            sut.Create(firstRequest, context);
+            
+            //assert
+            Assert.NotNull(capturedRequests);
+            Assert.Equal(requests, capturedRequests);
         }
 
         [Fact]
         public void SpecimenRequestsAreEmptyWhenAllThatWereRequestedAreCreated()
         {
-            // Fixture setup
-            var tracer = new DelegatingTracingBuilder();
-            var specimens = new[] { new object(), new object(), new object() };
-            var requestEvents = specimens.Select(
-                (o, i) => new RequestTraceEventArgs(o, i)).ToList();
-            var createdEvents = specimens.Reverse().Select(
-                (o, i) => new SpecimenCreatedEventArgs(o, null, i)).ToList();
-            var sut = new TerminatingWithPathSpecimenBuilder(tracer);
-            // Exercise system
-            requestEvents.ForEach(tracer.RaiseSpecimenRequested);
-            createdEvents.ForEach(tracer.RaiseSpecimenCreated);
-            // Verify outcome
+            //arrange
```

Comment:
Throughout the AutoFixture codebase, the [Four Phase Test](http://xunitpatterns.com/Four%20Phase%20Test.html) naming convention is used. Particularly, for existing tests, I don't see any reason to switch to [Arrange-Act-Assert](http://wiki.c2.com/?ArrangeActAssert)? 😃